<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1406</width>
    <height>928</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="font">
   <font>
    <family>Calibri</family>
    <pointsize>12</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>LetterShark</string>
  </property>
  <property name="styleSheet">
   <string notr="true">
	background-color: #1E1E1E;
	color: rgb(212, 212, 212);


</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1381</width>
      <height>901</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
      <horstretch>1</horstretch>
      <verstretch>1</verstretch>
     </sizepolicy>
    </property>
    <property name="styleSheet">
     <string notr="true">QTabWidget::pane {
    border: 1px solid #3C3C3C;      /* Rahmen um das Tab-Widget */
    background: #1E1E1E;            /* Inhalt-Hintergrund */
    top: -1px;                      /* damit der aktive Tab oben anstößt */
}

/* Die einzelnen Reiter */
QTabBar::tab {
    background: #2D2D30;            /* Inaktiver Tab */
    color: #9A9A9A;                 /* Inaktiver Text */
    padding: 6px 12px;
    margin-right: 1px;
    border-top-left-radius: 4px;
    border-top-right-radius: 4px;
    border: 1px solid #3C3C3C;      /* Rahmen an allen Seiten */
    border-bottom: none;            /* damit’s „in“ das Pane reinragt */
}

/* Hover-Effekt */
QTabBar::tab:hover {
    background: #094771;            /* dein Hover-Blau */
    color: #D4D4D4;
}

/* Der aktive Tab */
QTabBar::tab:selected {
    background: #1E1E1E;            /* Active Tab BG */
    color: #D4D4D4;                 /* Active Text */
    border: 1px solid #3C3C3C;
    border-bottom: 2px solid #007ACC; /* der blaue Strich unten */
}

/* Wenn du einen Tab disabled hast */
QTabBar::tab:disabled {
    background: #2D2D30;
    color: #5A5A5A;
}

/* Optional: Trennlinie zwischen TabBar und Inhalt verstecken */
QTabBar {
    border-bottom: none;
}</string>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <property name="tabsClosable">
     <bool>false</bool>
    </property>
    <widget class="QWidget" name="tw_overview">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <attribute name="title">
      <string>Overview</string>
     </attribute>
     <widget class="QWidget" name="gridLayoutWidget">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>50</y>
        <width>1001</width>
        <height>651</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="1" column="0" colspan="3">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="font">
            <font>
             <family>Calibri</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="styleSheet">
            <string notr="true">QLabel {
    color: #D4D4D4;        /* Primär-Text */
    background: transparent;
}</string>
           </property>
           <property name="text">
            <string>Search Phrase</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="le_search_phrase">
           <property name="font">
            <font>
             <family>Calibri</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="styleSheet">
            <string notr="true">QLineEdit {
    background: #1E1E1E;                /* BG */
    border: 1px solid #3C3C3C;          /* Border */
    color: #D4D4D4;                     /* Text */
    padding: 4px;                       /* etwas Innenabstand */
    selection-background-color: #094771;/* Markierung (optional) */
}

QLineEdit:focus {
    border: 1px solid #007ACC;          /* Focus Border */
    /* Hinweis: echtes Shadow-Highlight geht nur mit QGraphicsDropShadowEffect im Code */
}</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="cb_search_column">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="minimumSize">
            <size>
             <width>250</width>
             <height>28</height>
            </size>
           </property>
           <property name="font">
            <font>
             <family>Calibri</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="styleSheet">
            <string notr="true">QComboBox {
    background: #252526;               /* Closed BG */
    color: #D4D4D4;                    /* Closed Text */
    border: 1px solid #3C3C3C;         /* Rahmen passend zu den Inputs */
    padding: 3px 24px 3px 8px;         /* rechts Platz für den Pfeil */
    min-height: 20px;
    selection-background-color: #094771; /* markierter Eintrag im Edit-Feld */
}

QComboBox::drop-down {
    subcontrol-origin: padding;
    subcontrol-position: top right;
    width: 20px;
    border-left: 1px solid #3C3C3C;
    background: transparent;
}

QComboBox::down-arrow {
    image: url(C:/Users/dkueh/Workspace/cpp/log_viewer/src/images/down-arrow.png);
    width: 12px;
    height: 12px;
}

/* Das Popup-List-Widget */
QComboBox QAbstractItemView {
    background: #1E1E1E;                /* Open List BG */
    color: #D4D4D4;                     /* Open List Item Text */
    border: 1px solid #3C3C3C;
    selection-background-color: #094771; /* Open List Item Hover */
    padding: 2px;
    outline: 0;                         /* keinen blauen Rand */
}

/* Hover-Effekt in der Liste */
QComboBox QAbstractItemView::item:hover {
    background: #094771;                /* Open List Item Hover */
    color: #FFFFFF;
}

/* Ausgewählter Eintrag (wenn geschlossen bleibt er der Text in QComboBox) */
QComboBox QAbstractItemView::item:selected {
    background: #094771;
    color: #FFFFFF;
}

/* Disabled State */
QComboBox:disabled {
    background: #2D2D30;
    color: #5A5A5A;
}</string>
           </property>
           <property name="placeholderText">
            <string>Select File...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="2" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QCheckBox" name="chkb_global_search">
           <property name="styleSheet">
            <string notr="true">QCheckBox {
    spacing: 6px;                   /* Abstand zwischen Box und Text */
    color: #D4D4D4;                 /* Textfarbe */
}

/* Die Box selbst (Indicator) */
QCheckBox::indicator {
    width: 16px;
    height: 16px;
    border: 1px solid #3C3C3C;     /* Rahmen */
    border-radius: 3px;            /* leicht abgerundet */
    background: #1E1E1E;           /* Hintergrund der Box */
}

/* Hover über der Box */
QCheckBox::indicator:hover {
    border: 1px solid #007ACC;
}

/* Angekreuzt – Haken und Hintergrund */
QCheckBox::indicator:checked {
    background: #007ACC;            /* Accent als Fill */
    border: 1px solid #007ACC;
}
QCheckBox::indicator:checked:hover {
    background: #094771;            /* etwas dunkler beim Hover */
    border: 1px solid #094771;
}

QCheckBox::indicator:checked {
    image: url(C:/Users/dkueh/Workspace/cpp/log_viewer/src/images/check.png);
}
</string>
           </property>
           <property name="text">
            <string>Disable Global Search</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btn_search">
           <property name="font">
            <font>
             <family>Calibri</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="styleSheet">
            <string notr="true">QPushButton {
    background-color: #0E639C;      /* Primary BG */  
    color: #FFFFFF;                 /* Primary Text */  
    border: none;                   /* klarer, moderner Look */  
    padding: 2px 8px;              
    border-radius: 4px;             /* abgerundete Ecken */  
}

QPushButton:hover {
    background-color: #1177BB;      /* Primary Hover */  
}

QPushButton:pressed {
    background-color: #0C5A9A;      /* etwas dunkleres Blau */  
}

/* Optional: Disabled State */
QPushButton:disabled {
    background-color: #2D2D30;      /* geblendet, analog Secondary Disabled */  
    color: #5A5A5A;                 
}
</string>
           </property>
           <property name="text">
            <string>Search</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btn_reset">
           <property name="font">
            <font>
             <family>Calibri</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="styleSheet">
            <string notr="true">QPushButton {
    background-color: #3C3C3C;      /* Secondary BG */  
    color: #D4D4D4;                 /* Secondary Text */  
    border: none;                   
    padding: 2px 8px;                
    border-radius: 4px;             
}

QPushButton:hover {
    background-color: #505050;      /* Secondary Hover */  
}

QPushButton:pressed {
    background-color: #3A3A3A;      /* leicht dunkler als Normal */  
}

QPushButton:disabled {
    background-color: #2D2D30;      
    color: #5A5A5A;                 
}</string>
           </property>
           <property name="text">
            <string>Reset</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="0" column="2">
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Policy::Minimum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>180</width>
           <height>30</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="3" column="0" colspan="3">
        <widget class="QScrollArea" name="scrollArea">
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="scrollAreaWidgetContents">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>997</width>
            <height>496</height>
           </rect>
          </property>
          <widget class="QLabel" name="lbl_search_results">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>1001</width>
             <height>491</height>
            </rect>
           </property>
           <property name="font">
            <font>
             <family>Calibri</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="styleSheet">
            <string notr="true">QLabel {
    color: #007ACC;        /* Accent-Farbe für Links, wichtige Infos */
	background-color: rgb(37, 37, 38);
	border: 1px solid rgb(60, 60, 60);
	padding-left: 5px; 
	padding-top: 4px; 
    padding-bottom: 4px;
}</string>
           </property>
           <property name="text">
            <string/>
           </property>
          </widget>
         </widget>
        </widget>
       </item>
       <item row="0" column="1">
        <layout class="QFormLayout" name="formLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="font">
            <font>
             <family>Calibri</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="styleSheet">
            <string notr="true">QLabel {
    color: #D4D4D4;        /* Primär-Text */
    background: transparent;
}</string>
           </property>
           <property name="text">
            <string>Files Parsed</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="lbl_parsed_files">
           <property name="font">
            <font>
             <family>Calibri</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_2">
           <property name="font">
            <font>
             <family>Calibri</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="styleSheet">
            <string notr="true">QLabel {
    color: #D4D4D4;        /* Primär-Text */
    background: transparent;
}</string>
           </property>
           <property name="text">
            <string>Lines Parsed</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLabel" name="lbl_parsed_lines">
           <property name="font">
            <font>
             <family>Calibri</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_3">
           <property name="font">
            <font>
             <family>Calibri</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="styleSheet">
            <string notr="true">QLabel {
    color: #D4D4D4;        /* Primär-Text */
    background: transparent;
}</string>
           </property>
           <property name="text">
            <string>Status</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLabel" name="lbl_parsing_status">
           <property name="font">
            <font>
             <family>Calibri</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="0" column="0">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Policy::Minimum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>180</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QTreeWidget" name="tw_opened_files">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>50</y>
        <width>301</width>
        <height>661</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Calibri</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">QTreeWidget {
    background-color: rgb(37, 37, 38);  /* Basis-Hintergrund */
    color: #DDD; 
    border-color: #3c3c3c;                 /* Standard-Textfarbe */
}

/* sichtbare Auswahlfarbe */
QTreeWidget::item:selected {
    background-color: #2A2D2E;   /* dein Akzent-Dunkelblau */
    color: #007ACC;              /* kontrastierender Text */
}

/* optional: Hover-Effekt */
QTreeWidget::item:hover {
    background-color: #2A2D2E;   /* leicht aufgehellt beim Überfahren */
}</string>
      </property>
      <property name="editTriggers">
       <set>QAbstractItemView::EditTrigger::NoEditTriggers</set>
      </property>
      <property name="selectionBehavior">
       <enum>QAbstractItemView::SelectionBehavior::SelectItems</enum>
      </property>
      <column>
       <property name="text">
        <string>Name</string>
       </property>
      </column>
     </widget>
     <widget class="QLabel" name="lbl_logo">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>800</y>
        <width>219</width>
        <height>59</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="pixmap">
       <pixmap resource="resources.qrc">:/icons/src/images/LetterShark_logo_kl.png</pixmap>
      </property>
      <property name="scaledContents">
       <bool>true</bool>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionOpen_File">
   <property name="text">
    <string>Open File</string>
   </property>
  </action>
  <action name="actionAdd_new_File">
   <property name="text">
    <string>Add new File</string>
   </property>
  </action>
  <action name="actionOpen_Folder">
   <property name="text">
    <string>Open Folder</string>
   </property>
  </action>
  <action name="actionClear_Space">
   <property name="text">
    <string>Clear Space</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
